File: 01a._Data_Files/data_handling.py
import pandas as panda
import xmltodict
import yaml


def get_myself_from_file(file_path):
    if file_path.endswith('.csv'):
        data = panda.read_csv(file_path)
    elif file_path.endswith('.json'):
        data = panda.read_json(file_path)
    elif file_path.endswith('.xml'):
        with open(file_path, 'r') as file:
            data = xmltodict.parse(file.read())
    elif file_path.endswith('.yaml') or file_path.endswith('.yml'):
        with open(file_path, 'r') as file:
            data = yaml.safe_load(file)
    else:
        raise ValueError(f"Forkert filformat: {file_path}")

    return data

file_path_csv = 'me.csv'
file_path_json = 'me.json'
file_path_xml = 'me.xml'
file_path_yaml = 'me.yaml'

myself_from_csv = get_myself_from_file(file_path_csv)
myself_from_json = get_myself_from_file(file_path_json)
myself_from_xml = get_myself_from_file(file_path_xml)
myself_from_yaml = get_myself_from_file(file_path_yaml)

print("Mig fra CSV:" + '\n', myself_from_csv)
print('\n' + "Mig fra JSON:" + '\n', myself_from_json)
print('\n' + "Mig fra XML:" + '\n', panda.DataFrame(myself_from_xml))
print('\n' + "Mig fra YAML:" + '\n', panda.DataFrame([myself_from_yaml]))

File: 01a._Data_Files/data_handling.js
const fs = require('fs');
const Papa = require('papaparse');
const xml2js = require('xml2js');
const yaml = require('js-yaml');

//CSV fil
function readCSV(filePath) {
  const csvData = fs.readFileSync(filePath, 'utf8');
  
  Papa.parse(csvData, {
    header: true,
    complete: (result) => {
      console.log('Mig fra CSV:', result.data);
    },
    error: (error) => {
      console.error('Error fra CSV:', error.message);
    }
  });
}

//JSON fil
function readJSON(filePath) {
  const jsonData = fs.readFileSync(filePath, 'utf8');
  const data = JSON.parse(jsonData);

  console.log('Mig fra JSON:', data);
}

//XML fil
function readXML(filePath) {
  const xmlData = fs.readFileSync(filePath, 'utf8');
  xml2js.parseString(xmlData, (err, result) => {
    console.log('Mig fra XML:', result.me);
  });
}

//Yaml fil
function readYAML(filePath) {
  const yamlData = fs.readFileSync(filePath, 'utf8');
  const data = yaml.load(yamlData);

  console.log('Mig fra YAML:', data);
}


readCSV('me.csv');
readJSON('me.json');
readXML('me.xml');
readYAML('me.yaml');
